### PR TITLE
Remove check provision lane for 1.25 PSA as no longer needed

### PR DESCRIPTION
**Pull request by brianmcarey**
/cc  @xpivarc  @dhiller

### CODE REVIEW

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml [hunk 1]
```diff
@@ -264,32 +264,6 @@ presubmits:
...
-      preset-podman-in-container-enabled: "true"
-    max_concurrency: 2
-    name: check-provision-k8s-1.25-psa
-    spec:
-      containers:
-      - command:
-        - /usr/local/bin/runner.sh
-        - /bin/sh
-        - -c
-        - cd cluster-provision/k8s/1.25 && KUBEVIRT_PSA='true' ../provision.sh
```
Comment (xpivarc) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml:283]: Do we need to add `&& KUBEVIRT_PSA='true'` into `check-provision-k8s-1.25`?

Comment (brianmcarey) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml:283]: From a quick look - PSA is only enabled on the 1.26 kubevirt presubmits so I don't think we should enable it for the 1.25 provider check.

Comment (xpivarc) [reply github/ci/prow-deploy/files/jobs/kubevirt/kubevirtci/kubevirtci-presubmits.yaml:283]: We don't need to

**Pull request merged**